ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -1022,4 +1046,53 @@ public static String stripSlashAndExtensionFromId(String blobIdWithExtension) {
     int endIndex = extensionIndex != -1 ? extensionIndex : blobIdWithExtension.length();
     return blobIdWithExtension.substring(startIndex, endIndex);
   }
+
+  /**
+   * Class to encode decode kv header values.
+   */
+  static class KVHeaderValueEncoderDecoder {
+    private static String DELIM = "; ";
+    private static String KV_SEPERATOR = "=";
+
+    /**
+     * Encode a map of {@link String}s as http header value.
+     * @param map {@link Map} of {@link String}s to encode.
+     * @return encoded http header value.
+     */
+    static String encodeKVHeaderValue(Map<String, String> map) {
+      Objects.requireNonNull(map);
+      String value = "";
+      for (Map.Entry<String, String> entry : map.entrySet()) {
+        value += entry.getKey() + KV_SEPERATOR + entry.getValue() + DELIM;
+      }
+      return value;
+    }
+
+    /**
+     * Decode a kv header value.
+     * @param value {@link String} containing the encoded kv values.
+     * @return decoded kv {@link Map}.
+     */
+    static Map<String, String> decodeKVHeaderValue(String value) {
+      Objects.requireNonNull(value);
+      Map<String, String> valueMap = new HashMap<>();
+      if (value.isEmpty()) {
+        return valueMap;
+      }
+      if (!value.contains(DELIM)) {
+        throw new IllegalArgumentException("Invalid kv header value: " + value);
+      }
+      for (String kvStr : value.split(DELIM)) {
+        if (!kvStr.contains(KV_SEPERATOR)) {
```

Comment:
If the string ends in DELIM, won't split-ing it result in an extra empty string at the end of the array?